src/aggregate/aggregate_exec.c

```diff
@@ -223,20 +223,36 @@ static size_t serializeResult(AREQ *req, RedisModule_Reply *reply, const SearchR
         flags |= (req->reqflags & QEXEC_FORMAT_EXPAND) ? SENDREPLY_FLAG_EXPAND : 0;
 
         unsigned int apiVersion = req->sctx->apiVersion;
-        if (v && v->t == RSValue_Duo) {
-          // Which value to use for duo value
+        if (v && v->t == RSValue_JSON) {
+          // Which value to use for JSON value
           if (!(flags & SENDREPLY_FLAG_EXPAND)) {
             // STRING
             if (apiVersion >= APIVERSION_RETURN_MULTI_CMP_FIRST) {
               // Multi
-              v = RS_DUOVAL_OTHERVAL(*v);
+              if(!RS_JSONVAL_SERIALIZED(*v)) {
+                RedisModuleString *serialized;
+                japi->getJSONFromIter(RS_JSONVAL_ITER(*v), reply->ctx, &serialized);
```

Comment:
Is this thread safe? we might not have the GIL held at this point. What happens to an iterator if the JSON key is modified (update/delete)?